packages/rum-core/src/domain/rumEventsCollection/action/getSelectorsFromElement.ts

```diff
@@ -0,0 +1,136 @@
+import { arrayFrom, cssEscape, elementMatches } from '@datadog/browser-core'
+
+/**
+ * Stable attributes are attributes that are commonly used to identify parts of a UI (ex:
+ * component). Those attribute values should should not be generated randomly (hardcoded most of the
+ * time) and stay the same across deploys. They are not necessarily unique across the document.
+ */
+const STABLE_ATTRIBUTES = [
+  'data-dd-action-name',
+  // Common test attributes (list provided by google recorder)
+  'data-testid',
+  'data-test',
+  'data-qa',
+  'data-cy',
+  'data-test-id',
+  'data-qa-id',
+  'data-testing',
+  // Fullstory decorator attributes:
+  'data-component',
+  'data-element',
+  'data-source-file',
+]
+
+export function getSelectorsFromElement(element: Element, actionNameAttribute: string | undefined) {
+  const attributeSelectors = STABLE_ATTRIBUTES.map((attribute) => getAttributeSelector.bind(null, attribute))
+  if (actionNameAttribute) {
+    attributeSelectors.unshift(getAttributeSelector.bind(null, actionNameAttribute))
+  }
+  return {
+    selector: getSelectorFromElement(element, [getIDSelector], [getClassSelector]),
+    selector_with_stable_attributes: getSelectorFromElement(
+      element,
+      attributeSelectors.concat(getIDSelector),
+      attributeSelectors.concat(getClassSelector)
+    ),
+  }
+}
+
+type GetSelector = (element: Element) => string | undefined
+
+function getSelectorFromElement(
+  targetElement: Element,
+  globallyUniqueSelectorStrategies: GetSelector[],
+  uniqueAmongChildrenSelectorStrategies: GetSelector[]
+): string {
+  const targetElementSelector: string[] = []
+  let element: Element | null = targetElement
+
+  while (element && element.nodeName !== 'HTML') {
+    const uniqueSelector = findSelector(element, globallyUniqueSelectorStrategies, isSelectorUniqueGlobally)
+    if (uniqueSelector) {
+      targetElementSelector.unshift(uniqueSelector)
+      break
+    }
+
+    targetElementSelector.unshift(
+      findSelector(element, uniqueAmongChildrenSelectorStrategies, isSelectorUniqueAmongChildren) ||
+        getPositionSelector(element)
+    )
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏having a similar flow could ease the understanding
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
```suggestion
    const uniqueSelector = findSelector(element, globallyUniqueSelectorStrategies, isSelectorUniqueGlobally)
    if (uniqueSelector) {
      targetElementSelector.unshift(uniqueSelector)
      break
    }

    const uniqueSelectorAmongChildren = findSelector(element, uniqueAmongChildrenSelectorStrategies, isSelectorUniqueAmongChildren)
    if (uniqueSelectorAmongChildren) {
      targetElementSelector.unshift(uniqueSelectorAmongChildren)
    } else {
      targetElementSelector.unshift(getPositionSelector(element))
    }
```